rs/system_api/src/routing.rs

```diff
@@ -332,30 +328,30 @@ fn route_idkg_message(
 
     match requested_subnet {
         Some(subnet_id) => match network_topology.subnets.get(subnet_id) {
-            None => Err(ResolveDestinationError::IDkgKeyError(format!(
+            None => Err(ResolveDestinationError::ChainKeyError(format!(
                 "Requested threshold key {} from unknown subnet {}",
                 key_id, subnet_id
             ))),
             Some(subnet_topology) => {
                 if subnet_topology.idkg_keys_held.contains(key_id) {
-                    match idkg_subnet_kind {
-                        IDkgSubnetKind::HoldsAndSignWithKey => {
+                    match chain_key_subnet_kind {
+                        ChainKeySubnetKind::HoldsEnabledKey => {
                             if network_topology
                                 .idkg_signing_subnets(key_id)
```

Comment:
Eventually we should also rename this (`NetworkTopology::idkg_signing_subnets`) to, e.g., `NetworkTopology::chainkey_enabled_subnets`, but that can be done in another PR.

Reply:
Yes, I think doing that in a separate PR is better, as it involves renaming some protobuf types and fields (which requires disabling the `buf` check)